src/subdatatable/subdatatable.jl

```diff
@@ -90,36 +90,31 @@ function SubDataTable{T <: AbstractVector{Int}}(parent::DataTable, rows::T)
 end
 
 function SubDataTable(parent::DataTable, row::Integer)
-    return SubDataTable(parent, [row])
+    return SubDataTable(parent, [Int(row)])
 end
 
 function SubDataTable{S <: Integer}(parent::DataTable, rows::AbstractVector{S})
-    return view(parent, Int(rows))
+    return SubDataTable(parent, convert(Vector{Int}, rows))
 end
 
-
-function Base.view{S <: Real}(dt::DataTable, rowinds::AbstractVector{S})
-    return SubDataTable(dt, rowinds)
+function SubDataTable(parent::DataTable, rows::AbstractVector{Bool})
+    return SubDataTable(parent, find(rows))
 end
 
-function Base.view{S <: Real}(sdt::SubDataTable, rowinds::AbstractVector{S})
+function SubDataTable(sdt::SubDataTable, rowinds::Any)
```

Comment:
AFAICT this function is only supposed to be passed `AbstractVector{<:Integer}`, isn't it? If so, better restrict the signature to make the code clearer and get an error early.